destination/destination.go

```diff
@@ -0,0 +1,369 @@
+// Copyright © 2022 Meroxa, Inc. and Miquido
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package destination
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"sync"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/miquido/conduit-connector-elasticsearch/internal/elasticsearch"
+)
+
+func NewDestination() sdk.Destination {
+	return &Destination{}
+}
+
+type Destination struct {
+	sdk.UnimplementedDestination
+
+	config          Config
+	client          client
+	mutex           sync.Mutex
+	operationsQueue BufferQueue
+}
+
+const (
+	actionInsert  = "insert"
+	actionCreate  = "create"
+	actionCreated = "created"
+	actionUpdate  = "update"
+	actionUpdated = "updated"
+	actionDelete  = "delete"
+	actionDeleted = "deleted"
```

Comment:
It seems not to be standardized, so I wanted to support as many reasonable possibilities as possible.

Reply:
You mean, not standardized in the ES API?

Reply:
No, these are values from the Record's metadata["action"] field. So it actually depends on creators of source connectors 😉 I've seen you use action=delete for S3 and insert, update, delete for Postgres while Salesforce uses created, updated etc.

Reply:
Gotcha. We're working on getting standardized through [OpenCDC](https://medium.com/meroxa/a-proposal-for-better-interoperability-with-change-data-capture-13bec94aded1). So it would be good to mention the reason for the duplication in a comment.

Reply:
@donatorsky Is there a special reason why the Salesforce connector uses the "d" version? The S3 and Pg connectors are in line with the OpenCDC guidelines here: https://github.com/ConduitIO/conduit/blob/main/docs/design-documents/20220309-opencdc.md?plain=1#L267-L272

Reply:
It is because I get "d" version directly from the Salesforce in query response.

From this PR point of view, it seems I should drop "d" versions, go with OpenCDC guidelines and update Salesforce connector to translate SF ⇒ OpenCDC actions. Should I proceed with these steps?

Reply:
@donatorsky That's right, I'd go with that. Internally, connectors are free to do what they want, but the input and output should match the Conduit APIs and the OpenCDC guidelines (which are work in progress obviously and will be refined over time).

Reply:
And Salesforce counterpart:
miquido/conduit-connector-salesforce@f1d0dec